code_generation/quantity.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class Quantity:
+    def __init__(self, name):
+        self.name = name
+        self.shifts = {}
+        self.children = {}
+        self.output_scopes = []
+        log.debug("Setting up new Quantity {}".format(self.name))
+
+    # the scopes, in which a quantity is used as an output is tracked in the output_scopes list.
+    # This check is triggered for every producer.
+    # If a quantity is already used within a given scope as output, this will result in an exception.
+    def check_scope(self, scope):
+        log.debug("Checking {} / scope {}".format(self.name, scope))
+        if scope not in self.output_scopes:
```

Comment:
you should also check whether it global is present

Reply:
but global is also just a scope or am i misunderstanding your comment ?